File: app/models/qdm/prediction.rb
module QDM
  # app/models/qdm/prediction.rb
  class Prediction < DataElement
    include Mongoid::Document
    embedded_in :patient
    field :outcome
    field :probability
    field :qualitativeRisk
    field :relativeRisk
    field :when
    field :rationale
    field :qdmVersion, type: String, default: '4.0.0'
  end
end

File: app/assets/javascripts/Prediction.js
const mongoose = require('mongoose');
const { DataElementSchema } = require('./basetypes/DataElement');
const Code = require('./basetypes/Code');
const Interval = require('./basetypes/Interval');
const Quantity = require('./basetypes/Quantity');
const DateTime = require('./basetypes/DateTime');
const { ComponentSchema } = require('./Component');
const { FacilityLocationSchema } = require('./FacilityLocation');
const { IdSchema } = require('./Id');
const Any = require('./basetypes/Any');

const [Number, String] = [
  mongoose.Schema.Types.Number,
  mongoose.Schema.Types.String,
];

const PredictionSchema = DataElementSchema({
  outcome: Any,
  probability: Any,
  qualitativeRisk: Any,
  relativeRisk: Any,
  when: Any,
  rationale: Any,
  qdmVersion: { type: String, default: '4.0.0' },
  _type: { type: String, default: 'Prediction' },

});

module.exports.PredictionSchema = PredictionSchema;
module.exports.Prediction = mongoose.model('Prediction', PredictionSchema);
